### PR TITLE
Add accounting data for collSurplus allocations

### PR DESCRIPTION
**Pull request by dapp-whisperer**
Accounting event tweak to solve #729 for web2 team.

Small event change involving forwarding of information, given that we're doing notable audit mitigations it seemed appropriate so fit in this request.

# CollSurplusPool

`collSurplusPool.increaseSurplusCollShares` now forwards along CdpId for the event, and splits the collSurplus to add between two variables, `collateralShares` and `liquidatorRewardShares`

`collateralShares` and `liquidatorRewardShares` are both added to the total new claimable balance for the user.

```solidity
function increaseSurplusCollShares(
        bytes32 _cdpId,
        address _account,
        uint256 _collateralShares,
        uint256 _liquidatorRewardShares
    ) external;
```

This is ultimately to add addition data to this event, which is also renamed to show it is only for increases.
```solidity
event SurplusCollSharesAdded(
        bytes32 indexed _cdpId,
        address indexed _account,
        uint256 _claimableSurplusCollShares,
        uint256 _surplusCollSharesAddedFromCollateral,
        uint256 _surplusCollSharesAddedFromLiquidatorReward
    );
```

It no longer plays double-duty with the "claimed" event which was added / slightly renamed.

```solidity
event SurplusCollSharesClaimed(address indexed _account, uint256 _collShares);
```

we emit both that and `event CollSharesTransferred(address indexed _to, uint256 _amount);` to fit the paradigm established by the activePool.

This is a waste of gas to duplicate so will sync with @mrbasado to determine utility.

Needs some simple tests still

### CODE REVIEW

File: packages/contracts/contracts/Dependencies/RolesAuthority.sol [hunk 1]
```diff
@@ -56,7 +56,7 @@ contract RolesAuthority is IRolesAuthority, Auth, Authority {
     //////////////////////////////////////////////////////////////*/
 
     /**
-        @notice A user can call a given function signature on a given target address if:
+        @notice A user can c    all a given function signature on a given target address if:
```
Comment (rayeaster) [inline packages/contracts/contracts/Dependencies/RolesAuthority.sol:59]: typo?

File: packages/contracts/contracts/Interfaces/ICollSurplusPool.sol [hunk 2]
```diff
@@ -5,7 +5,14 @@ pragma solidity 0.8.17;
 interface ICollSurplusPool {
     // --- Events ---
 
-    event SurplusCollSharesUpdated(address indexed _account, uint256 _newBalance);
+    event SurplusCollSharesClaimed(address indexed _account, uint256 _collShares);
```
Comment (rayeaster) [inline packages/contracts/contracts/Interfaces/ICollSurplusPool.sol:8]: does it make sense to add before and after surplus value for `_account`?

**Review by rayeaster — APPROVED** [review]
LGTM

File: packages/contracts/contracts/Interfaces/ICollSurplusPool.sol (hunk 2 above)
Comment (GalloDaSballo) [reply packages/contracts/contracts/Interfaces/ICollSurplusPool.sol:8]: It's always 100% and they claim all, so the new value will be 0

File: packages/contracts/contracts/CollSurplusPool.sol [hunk 3]
```diff
@@ -101,6 +116,7 @@ contract CollSurplusPool is ICollSurplusPool, ReentrancyGuard, AuthNoOwner {
         unchecked {
             totalSurplusCollShares = cachedTotalSurplusCollShares - claimableColl;
         }
+        emit SurplusCollSharesClaimed(_account, claimableColl);
```
Comment (GalloDaSballo) [inline packages/contracts/contracts/CollSurplusPool.sol:119]: Do we need both events? They are basically the same

Comment (dapp-whisperer) [reply packages/contracts/contracts/CollSurplusPool.sol]: removed this and sticking with the `CollSharesTransferred` event

File: packages/contracts/contracts/Interfaces/ICollSurplusPool.sol (hunk 2 above)
Comment (dapp-whisperer) [reply packages/contracts/contracts/Interfaces/ICollSurplusPool.sol:8]: removing this event in favor of identical `CollSharesTransferred`

**Review by GalloDaSballo — APPROVED** [review]
The PR chunks up the values passed to `increaseSurplusCollShares`
Then emits an event with them

The other event `SurplusCollSharesUpdated` was unnecessary and was removed

UI can use 
- Claims -> `CollSharesTransferred` (any transfer + removal of the surplus when from is `CollSurplus`
- Coll surplus add -> `SurplusCollSharesAdded`

**Pull request merged**